src/components/eventListeners.js

```diff
@@ -1,68 +1,133 @@
 import {
-  state,
-  SEARCH_TEXT_INPUT,
-  SUGGESTION_LIST,
-  RECENT_SEARCHES_LIST,
-  MY_LOCATION_BUTTON,
-  GO_BUTTON,
-} from '../index';
-import { renderSuggestions, renderSearchList } from './view';
-import { getSuggestions } from './api';
-
-function onFocus() {
+  searchTextInput,
+  suggestionList,
+  recentSearchList,
+  myLocationButton,
+  goButton,
+  loadMoreButton,
+  propertyList,
+  searchContainer,
+  propertyContainer,
+  backButton,
+  propertySpinnerContainer,
+  propertySpinnerCanvas,
+} from './domConsts';
+import { state } from '../index';
+import { renderSuggestions, renderSearchList, renderProperties } from './view';
+import { getSuggestions, getProperties } from './api';
+import Spinner from './spinner';
+
+function onSearchFocus() {
   state.showSuggestions = true;
   renderSuggestions();
 }
 
-async function onInput(e) {
-  await getSuggestions(e.target.value);
-  renderSuggestions();
-}
+const onSearchInput = (() => {
+  // 1 sec delay before fetching data
+  let timeout;
 
-function onBlur() {
-  SUGGESTION_LIST.style.height = '0px';
+  return e => {
+    try {
+      const suggest = async () => {
+        await getSuggestions(e.target.value);
+        renderSuggestions();
+      };
+      if (timeout) clearTimeout(timeout);
+      timeout = setTimeout(suggest, 1000);
+    } catch (e) {
+      console.log(e);
+    }
+  };
+})();
+
+function onSearchBlur() {
+  suggestionList.style.height = '0px';
   state.showSuggestions = false;
 }
 
 function onSuggestionClick(e) {
-  state.currentSearchItem = state.suggestions.find(
-    item => item._id === e.target.dataset.key
+  if (e.target.nodeName !== 'LI') return;
+
+  const index = Array.from(suggestionList.children).findIndex(
+    item => item.outerText === e.target.innerText
   );
-  SEARCH_TEXT_INPUT.value = `${state.currentSearchItem.area_type}, ${state.currentSearchItem.city}, ${state.currentSearchItem.state_code}`;
+
+  state.currentSearchItem = state.suggestions[index];
+  const { area_type, city, state_code } = state.currentSearchItem;
+  searchTextInput.value = `${area_type}, ${city}, ${state_code}`;
 }
-function onMyLocationClick() {
+
+function onMyLocationButtonClick() {
   // get random location from suggestion list
-  state.currentSearchItem =
-    state.suggestions[Math.floor(Math.random() * state.suggestions.length)];
-  SEARCH_TEXT_INPUT.value = `${state.currentSearchItem.area_type}, ${state.currentSearchItem.city}, ${state.currentSearchItem.state_code}`;
+  state.currentSearchItem = state.suggestions[Math.floor(Math.random() * state.suggestions.length)];
+  searchTextInput.value = `${state.currentSearchItem.area_type}, ${state.currentSearchItem.city}, ${state.currentSearchItem.state_code}`;
```

Comment:
use:
const { area_type, city, state_code } = state.currentSearchItem;
searchTextInput.value = `${area_type}, ${city}, ${ state_code}`,

I would be better to create getter:
get<Yourname>Value() {
  const { area_type, city, state_code } = state.currentSearchItem;
  return `${area_type}, ${city}, ${ state_code}`,
}

function onSuggestionClick() {
 //...
  searchTextInput.value = get<Yourname>Value()
}